Regular Expressions/File Pattern.sublime-syntax

```diff
@@ -0,0 +1,38 @@
+%YAML 1.2
+---
+# https://www.sublimetext.com/docs/file_patterns.html
+name: Sublime File Pattern
+file_extensions:
+  - sublime-file-patterns
+hidden: true
+scope: source.file-pattern
+
+contexts:
+  main:
+    - include: multiple
+
+  multiple:
+    - match: (,)\s*(?:(-)|(//))?
+      captures:
+        1: punctuation.separator.file-pattern
+        2: keyword.operator.logical.file-pattern
+        3: constant.language.file-pattern
+    - include: single
+
+  single:
+    - match: \*{2,}
+      scope: invalid.illegal.file-pattern
+    - match: \[|\]
+      scope: invalid.illegal.file-pattern
+    - match: ^\s*(?:(-)|(//))
```

Comment:
This won't match project-relative patterns at any position that is not the first, or exclusive patterns for that matter because of the `^` anchor. You'll need to restructure the to match certain characters only at the first position, for example through a multi-push on `(?=[^\s,])`. (I'm not sure how white-space-sensitive patterns are.)

Of course, you should also add tests for these cases.

Reply:
How about now?